tests/testthat/test-infix_spaces_linter.R

```diff
@@ -2,15 +2,19 @@ test_that("returns the correct linting", {
   ops <- c(
     "+",
     "-",
+    "~",
     "=",
     "==",
     "!=",
     "<=",
     ">=",
     "<-",
+    ":=",
+    "<<-",
     "<",
     ">",
     "->",
+    "->>",
```

Comment:
I just noticed that the test suite tests `1 <- 2` etc.
IMO we should add sensible test cases for the assignment operators.

Reply:
I copy-pasted the tests for the deleted `assignment_spaces_linter` here. I also removed a bunch of them that were throwing lints for using >1 space, because the style guide says that's OK _if it's used to improve alignment_ and our logic is (and I think should continue to be) not sophisticated enough to detect alignment issues:

https://style.tidyverse.org/syntax.html#extra-spaces

> Adding extra spaces is ok if it improves alignment of `=` or `<-`.

Reply:
Since @f-ritter's original use-case was disallowing these, I think we should also support that rule via an argument (`allow_multiple_spaces = TRUE`). WDYT?

Reply:
Okay, should be a lot easier if we convert the linter to XPath at the same time (requires something like `following-sibling::*[@col1 = preceding-sibling::*/@col2]` vs. `> preceding-sibling::*/@col2` iinm)